From ac8debcedda20a7713bfa13411d7cd916dc5295a Mon Sep 17 00:00:00 2001
From: Julien Malik <julien.malik@paraiso.me>
Date: Wed, 14 Mar 2018 19:20:20 +0100
Subject: [PATCH 14/15] iio_adi_xflow_check: Avoid accessing device after
 context destruction

The monitor thread must be joined before the call
to iio_context_destroy, otherwise iio_device_reg_read
and iio_device_reg_write might be called after context
destruction.

Signed-off-by: Julien Malik <julien.malik@paraiso.me>
---
 tests/iio_adi_xflow_check.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tests/iio_adi_xflow_check.c b/tests/iio_adi_xflow_check.c
index 7d69888..a1d4be8 100644
--- a/tests/iio_adi_xflow_check.c
+++ b/tests/iio_adi_xflow_check.c
@@ -339,10 +339,10 @@ int main(int argc, char **argv)
 		}
 	}
 
+	pthread_join(monitor_thread, NULL);
+
 	iio_buffer_destroy(buffer);
 	iio_context_destroy(ctx);
 
-	pthread_join(monitor_thread, NULL);
-
 	return 0;
 }
-- 
2.11.0

